modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -196,12 +185,13 @@ public CompletableFuture<Void> onUpdate(WatchEvent event) {
                     LeaseTracker.this.leases = new Leases(unmodifiableMap(leasesMap), leasesBytes);
                 }
 
-                return completedFuture(null);
+                return allOf(fireEventFutures.toArray(CompletableFuture[]::new));
```

Comment:
Do we really we really won't to block ms notification thread with onBecomePrimary handling logic?

Reply:
I don’t quite understand your question, if you’re talking about **ReplicaListener#onBecomePrimary**, then we’ll get rid of this method soon. Well, or I completely dont understand why you want to block the metastor for **ReplicaListener#onBecomePrimary**.

Reply:
Never mind.